### ISSUE TITLE
Building feature at 2 zoom levels shown at once

### ISSUE THREAD

**Issue by nvkelso** [tangrams/tangram-es#411]
This is [McKinleyville High School](https://www.openstreetmap.org/#map=17/40.95529/-124.10899) in McKinleyville California. I only expected geometry from a single zoom level to be visible at once.

![screenshot_2015-11-28-20-22-14](https://cloud.githubusercontent.com/assets/853051/11494489/b9004f48-97b7-11e5-9118-04e27f7bfe9e.png)

**Comment by hjanetzek** [comment]
This may happen when a parent tile is still drawn while some of its children (the tiles matching the current zoom-level) are already loaded. The coarser pattern in the upper-right corner seems to still use the parent tile here. If the view remains as on the screenshot it could be that the tile is missing or failed to load

**Comment by tallytalwar** [comment]
@hjanetzek correctly described this behavior.

We have something called as `proxy` tiles in es, which are basically already loaded (parent or children) tiles from last update. 
These proxy tiles are drawn till the new actual visible tiles (respective children or parent tile for these proxies) are loaded and ready to be drawn.

Possible in the above instance, one of the children tile fail to load and you were seeing the parent tile along with other 3 children tiles, hence mixed geometries.

**Comment by bcamper** [comment]
But my understanding of desired proxy tile behavior is that the proxy
should be drawn underneath, so that the desired zoom level is always shown.
They should not be mixed.

**Comment by nvkelso** [comment]
That's also my expected behavior (proxy tiles drawn under).

**Comment by tallytalwar** [comment]
Yup proxy tiles are drawn under. Parent proxy tile is only visible for the not loaded child tile. Rest all loaded children tile should have overlapping proxy parent.

**Comment by hjanetzek** [comment]
Ah, yes, the proxy building should be drawn below the earth polygon of the current tile - but to enforce this we would need to know offset range to sweep the proxy under the carpet :) I guess for now we need to find a working building 'order' and should not tweak position.z in the custom building shader

**Comment by tallytalwar** [comment]
Proxy tiles are drawn deeper in depth buffer based on their z id.
https://github.com/tangrams/tangram-es/blob/master/core/resources/shaders/polygon.vs#L98-L100

I am checking as to why in this specific case, it appears that proxy tiles are drawn above visible tile set, and hence overlapping geometry.
